### ISSUE TITLE
Add a page listing all available docs versions

### ISSUE THREAD

**Issue by mbrukman** [JanusGraph/old-docs.janusgraph.org#22]
Right now, http://docs.janusgraph.org simply redirects to the latest set of docs.

We should add an index page which provides a list of all available doc sets, including past releases, and the upcoming future releases (typically a `SNAPSHOT` set of docs).

This will provide users with an easy-to-find list of available docs versions. Currently, the only way to find the set of available docs versions is to look at the available directories in the GitHub repo: https://github.com/JanusGraph/docs.janusgraph.org which is not easily discoverable.

-----

Once this is done, the "Other versions" header at the top of the page should point to link to this page, since "Other versions" next to "Documentation (0.1.0)" implies other _documentation_ versions, not other binary versions.

Right now, it points to the GitHub releases page, so it's just a duplicate of the "Releases" link that's already next to it.

**Comment by chupman** [comment]
@mbrukman rather than a separate page with 4-6 bullets would you be open to instead just making other versions a dropdown? Ideally it would allow you to switch between versions and stay on a page. 

So if you were on http://docs.janusgraph.org/latest/configuration.html and selected 0.1.0 from the dropdown you'd be taken to http://docs.janusgraph.org/0.1.0/configuration.html.
In the event that a version is selected where the current doc doesn't exist, like http://docs.janusgraph.org/0.1.0/configuredgraphfactory.html, we'd probably want some .htaccess rules that redirect to the index for that version.

**Comment by chupman** [comment]
@mbrukman Just opened a PR for this https://github.com/JanusGraph/janusgraph/pull/1046

preview can be found on my [fork](https://chupman.github.io/docs.janusgraph.org/latest/index.html) 
I just have the links going to index.html instead of switching between versions of whatever doc page you're on like I suggested in my previous comment.

**Comment by mbrukman** [comment]
@chupman — awesome work! Looks great. Please see my comment on the PR.

**Comment by porunov** [comment]
@mbrukman @chupman I think this issue is resolved. Isn't it?

**Comment by chupman** [comment]
@porunov you are correct.
